src/store/reducers/todo.js

```diff
@@ -21,6 +27,15 @@ export const todoReducer = (state = initialTodoState, action) => {
       return { ...state, tasks: filteredTasks };
     }
 
+    case MARK_TASK_DONE: {
+      const existingTasks = [...state.tasks];
+      const selectedTaskId = findTaskIndexById(action.payload, existingTasks);
```

Comment:
finding index of a collection doesn't mutate, copy is unnecessary. 

```suggestion
      const selectedTaskId = findTaskIndexById(action.payload, state.tasks);
```